getMacAddress() fails after boot

**Describe the bug**
`NetworkUtils.getMacAddress()` is failing after boot with the following error:

```
Dec 14 00:42:51 photonvision java[1356]: [2025-12-14 00:42:51] [General - NetworkUtils] [ERROR] Error getting MAC address: Cannot invoke "java.net.NetworkInterface.getHardwareAddress()" because the return value of "java.net.NetworkInterface.getByName(String)" is null
Dec 14 00:42:51 photonvision java[1356]: [2025-12-14 00:42:51] [General - NetworkUtils] [ERROR] java.lang.NullPointerException: Cannot invoke "java.net.NetworkInterface.getHardwareAddress()" because the return value of "java.net.NetworkInterface.getByName(String)" is null
Dec 14 00:42:51 photonvision java[1356]:         at org.photonvision.common.networking.NetworkUtils.getMacAddress(NetworkUtils.java:263)
Dec 14 00:42:51 photonvision java[1356]:         at org.photonvision.common.dataflow.networktables.NetworkTablesManager.checkHostnameAndCameraNames(NetworkTablesManager.java:253)
Dec 14 00:42:51 photonvision java[1356]:         at org.photonvision.common.util.TimedTaskManager$CaughtScheduledThreadPoolExecutor.lambda$wrap$0(TimedTaskManager.java:56)
Dec 14 00:42:51 photonvision java[1356]:         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
Dec 14 00:42:51 photonvision java[1356]:         at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
Dec 14 00:42:51 photonvision java[1356]:         at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
Dec 14 00:42:51 photonvision java[1356]:         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
Dec 14 00:42:51 photonvision java[1356]:         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
Dec 14 00:42:51 photonvision java[1356]:         at java.base/java.lang.Thread.run(Thread.java:840)
```

**Platform:**
 - Hardware Platform (ex. Raspberry Pi 4, Windows x64): Rubik Pi
   - How is it powered? (ex. Zinc-V, Pololu Buck Converter, Battery Bank): USB-C PD power supply
 - Network Configuration (Connection between the Radio and any devices in between, such as a Network Switch): Network Switch
 - PhotonVision Version:  dev-v2026.0.1-beta-1-g07affd4f
 - Browser (with Version) (Chrome, Edge, Firefox, etc.): Edge
 - Camera(s) Used: OV9281

**Additional context**
I tested by adding a loop that retries this the `NetworkInterface.getByName()` call until it doesn't return null. The call finally succeeded about 5 seconds after it was first tried.

It seems that, on Linux, [Java only recognizes "configured" interfaces i.e. interfaces with IP addresses](https://stackoverflow.com/a/17706515/7517724). Logging `nmcli dev status` after each attempt at `getByName()` reveals that the method returns null when the STATE is "connecting (getting IP configuration)" and it succeeds when the STATE is "connected".

--- a/photon-core/src/main/java/org/photonvision/common/networking/NetworkUtils.java
+++ b/photon-core/src/main/java/org/photonvision/common/networking/NetworkUtils.java
@@ -260,11 +260,14 @@ public class NetworkUtils {
                     }
                 }
             } else { // Managed? We should have a working interface available
-                byte[] mac = NetworkInterface.getByName(config.networkManagerIface).getHardwareAddress();
-                if (mac != null) {
-                    return formatMacAddress(mac);
-                } else {
-                    logger.error("No MAC address found for " + config.networkManagerIface);
+                var iface = NetworkInterface.getByName(config.networkManagerIface);
+                if (iface != null) {
+                    byte[] mac = iface.getHardwareAddress();
+                    if (mac != null) {
+                        return formatMacAddress(mac);
+                    } else {
+                        logger.error("No MAC address found for " + config.networkManagerIface);
+                    }
                 }
             }
         } catch (Exception e) {

